components/brave_private_new_tab_ui/resources/page/hooks.ts

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react'
+import getPageHandlerInstance from './api/brave_page_handler'
+import { loadTimeData } from '../../../common/loadTimeData'
+
+export function useTorObserver () {
+  const [hasInit, setHasInit] = React.useState(false)
+  const [isConnected, setIsConnected] = React.useState(false)
+  const [progress, setProgress] = React.useState<string | undefined>(undefined)
+
+  const isLoading = !hasInit && !isConnected
+
+  React.useEffect(() => {
+    getPageHandlerInstance().callbackRouter.onTorCircuitEstablished.addListener(setHasInit)
+    getPageHandlerInstance().callbackRouter.onTorInitializing.addListener(setProgress)
+  }, [])
+
+  React.useEffect(() => {
+    getPageHandlerInstance().pageHandler.getIsTorConnected().then((res: { isConnected: boolean }) => setIsConnected(res.isConnected))
+  }, [hasInit])
+
+  return {
+    isConnected,
+    isLoading,
+    progress
+  }
+}
+
+export function useHasSeenDisclaimer () {
+  const [hasSeenDisclaimer, setHasSeenDisclaimer] = React.useState<boolean>(loadTimeData.getBoolean('hasUserSeenDisclaimer'))
```

Comment:
@petemill When you close the disclaimer box for the first time, reload the page, there's a flash of the disclaimer box on subsequent reloads. Unless, if you destroy the window and open it again.

I wonder if the default here should always be false which matches with the default of user pref?

Reply:
Well, should it be the other way round to avoid this? In other words, do not show the "disclaimer" by default. And then if the API returns the decision that we should show it then we show it. Better to have it appear slightly after page load than flash visible then hidden, right? And then you can probably just remove the loadTimeData property. 

For times where we really do want this to be available at load-time and not wait for an async mojom call, there is a way around this: using `render_frame_host->SetWebUIProperty` in the Browser and `chrome.getVariableValue` in JS. And I believe that's updateable on a refresh. But I don't suggest it for this case.